packages/mobile/src/escrow/ReclaimPaymentConfirmationScreen.test.tsx

```diff
@@ -59,104 +84,56 @@ describe('ReclaimPaymentConfirmationScreen', () => {
         <ReclaimPaymentConfirmationScreen {...mockScreenProps} />
       </Provider>
     )
 
-    // Initial render should not contain a fee.
-    expect(toJSON()).toMatchSnapshot()
-    expect(queryAllByText('securityFee')).toHaveLength(2)
-    expect(queryByText(/-\s*0\.01/s)).toBeNull()
-
-    // Wait for fee to be calculated and displayed as "0.01"
-    // NOTE: Use regex here because the text may be split by a newline.
-    await waitFor(() => getByText(/-\s*0\.01/s))
-    expect(toJSON()).toMatchSnapshot()
-    // Query for the total amount, which should deduct the fee.
-    expect(queryByText(/₱\s*13\.28/s)).not.toBeNull()
+    expect(getElementText(getByTestId('LineItemRow/ReclaimAmount'))).toBe('₱13.30')
+    expect(getElementText(getByTestId('ReclaimFee'))).toBe('₱0.067')
+    expect(getElementText(getByTestId('TotalLineItem/Total'))).toBe('₱13.23')
+    expect(getElementText(getByTestId('TotalLineItem/Subtotal'))).toBe('9.95 cUSD')
+    expect(getByTestId('ConfirmButton')).not.toBeDisabled()
   })
 
   it('renders correctly when fee calculation fails', async () => {
-    mockedGetReclaimEscrowFee.mockImplementation(async () => {
-      throw new Error('Calculate fee failed')
-    })
-
-    const { queryAllByText, queryByText, getByText, toJSON } = render(
-      <Provider store={store}>
+    const { getByTestId, queryByTestId } = render(
+      <Provider store={store(mockFeeEstimate({ error: true }))}>
         <ReclaimPaymentConfirmationScreen {...mockScreenProps} />
       </Provider>
     )
 
-    // Initial render
-    expect(toJSON()).toMatchSnapshot()
-    expect(queryAllByText('securityFee')).toHaveLength(2)
-    expect(queryByText('₱0.001')).toBeNull()
-    expect(queryAllByText('$10.00')).toHaveLength(1)
-
-    // Wait for fee error
-    await waitFor(() => getByText('---'))
-
-    expect(queryAllByText('$10.00')).toHaveLength(1)
-    expect(toJSON()).toMatchSnapshot()
+    expect(getElementText(getByTestId('LineItemRow/ReclaimAmount'))).toBe('₱13.30')
+    expect(queryByTestId('ReclaimFee')).toBeFalsy()
+    expect(getElementText(getByTestId('LineItemRow/SecurityFee'))).toBe('---')
+    expect(getElementText(getByTestId('TotalLineItem/Total'))).toBe('₱13.30')
+    expect(getElementText(getByTestId('TotalLineItem/Subtotal'))).toBe('10.00 cUSD')
+    expect(getByTestId('ConfirmButton')).toBeDisabled()
   })
 
   it('shows the activity indicator when a reclaim is in progress', async () => {
-    mockedGetReclaimEscrowFee.mockImplementation(async () => TEST_FEE_INFO_CUSD)
-
     const tree = render(
-      <Provider store={store}>
+      <Provider store={store(mockFeeEstimate({ loading: true }))}>
         <ReclaimPaymentConfirmationScreen {...mockScreenProps} />
       </Provider>
     )
 
     expect(tree.UNSAFE_queryByType(ActivityIndicator)).toBeTruthy()
   })
-
-  it('clears the activity indicator when a reclaim fails', async () => {
```

Comment:
Well, the test isn't really working. It's checking that the activity indicator is falsey but not making it show up in the first place (ie, never calling the `mockedReclaimPayment`). It should press the confirm button first or something, otherwise it's just a false positive. 

I didn't think that the test was particularly useful too, since the behavior it's supposed to be testing is of the saga (the saga should store in the redux store that it's not reclaiming anymore).

In general, I think this part is implemented a bit strangely, the UI is try/catching the dispatch call which doesn't make much sense I think. I didn't want to spend the time going too in-depth into it, but I can spend some time cleaning it up if you prefer.

Reply:
Ah I see, probably a relic from when we had thunks 👍